reframe/frontend/executors/__init__.py

```diff
@@ -29,9 +92,13 @@ def __init__(self, check, listeners=[]):
         # Test case has finished, but has not been waited for yet
         self.zombie = False
 
+    @property
+    def testcase(self):
+        return self._case
+
     @property
     def check(self):
-        return self._check
+        return self._case.check
```

Comment:
@vkarak, I did not understand the concept...
Is the `case` part of the `check` or the `check` part of the `case`?

Reply:
The check is part of the test case. Here you are looking into the `RegressionTask`, which is a different thing. See my explanations at the general comments.